1111-i386-txt-Add-Intel-TXT-ACM-module-support.patch

```diff
@@ -0,0 +1,625 @@
++    {
++      grub_error (GRUB_ERR_OUT_OF_RANGE, N_("ACM header length and scratch size in bytes overflows"));
++      return NULL;
++    }
++
++  ptr = fits_in_acm(hdr, user_area_off, info_table_size);
++
++  if ( ptr != NULL )
++    {
++      if ( info_table_get (ptr, length) < info_table_size )
++	return NULL;
++
++      info_table_size = info_table_get (ptr, length);
++      ptr = fits_in_acm(hdr, user_area_off, info_table_size);
++    }
++
++  return ptr;
++}
++
++/*
++ * Function returns pointer to chipset ID list, after checking that
++ * grub_txt_acm_chipset_id_list and all grub_txt_acm_chipset_id structures are
++ * within ACM. Otherwise, NULL is returned.
++ */
++static struct grub_txt_acm_chipset_id_list*
++get_acmod_chipset_list (struct grub_txt_acm_header *hdr)
++{
++  struct grub_txt_acm_info_table *info_table;
++  grub_uint32_t id_entries_off;
++  struct grub_txt_acm_chipset_id_list *chipset_id_list;
++
++  /* This fn assumes that the ACM has already passed the is_acmod() checks */
++
++  info_table = get_acmod_info_table (hdr);
++  if ( info_table == NULL )
++    {
++      grub_error (GRUB_ERR_OUT_OF_RANGE, N_("ACM info table out of bounds"));
++      return NULL;
++    }
++
++  chipset_id_list = fits_in_acm(hdr, info_table_get (info_table, chipset_id_list),
++				sizeof(struct grub_txt_acm_chipset_id_list));
++  if ( chipset_id_list == NULL )
++    {
++      grub_error (GRUB_ERR_OUT_OF_RANGE, N_("ACM chipset ID list out of bounds"));
++      return NULL;
++    }
++
++  /* Overflows? */
++  if ( grub_add (info_table->chipset_id_list,
++		 sizeof(struct grub_txt_acm_chipset_id_list), &id_entries_off))
++    {
++      grub_error (GRUB_ERR_OUT_OF_RANGE, N_("ACM chipset ID entries out of bounds"));
++      return NULL;
++    }
++
++  if ( n_fit_in_acm( hdr, id_entries_off, sizeof(struct grub_txt_acm_chipset_id),
++		      chipset_id_list->count ) == NULL )
++    {
++      grub_error (GRUB_ERR_OUT_OF_RANGE, N_("ACM chipset ID entries out of bounds"));
++      return NULL;
++    }
++
++  return chipset_id_list;
++}
++
++/*
++ * Function returns pointer to processor ID list, after checking that
++ * grub_txt_acm_processor_id_list and all grub_txt_acm_processor_id structures
++ * are within ACM. Otherwise, NULL is returned.
++ */
++static struct grub_txt_acm_processor_id_list*
++get_acmod_processor_list (struct grub_txt_acm_header* hdr)
++{
++  struct grub_txt_acm_info_table *info_table;
++  grub_uint32_t id_entries_off;
++  struct grub_txt_acm_processor_id_list *proc_id_list;
++
++  /* This fn assumes that the ACM has already passed the is_acmod() checks */
++
++  info_table = get_acmod_info_table(hdr);
++  if ( info_table == NULL )
++    {
++      grub_error (GRUB_ERR_OUT_OF_RANGE, N_("ACM info table out of bounds"));
++      return NULL;
++    }
++
++  proc_id_list = fits_in_acm(hdr, info_table_get (info_table, processor_id_list),
++			     sizeof(struct grub_txt_acm_processor_id_list));
```

Comment:
```suggestion
+			     sizeof(*proc_id_list));
```

`sizeof()` does not evaluate its argument (unless the argument is of variably-modified type, which is not the case here).  Therefore, this is correct and type-safe.